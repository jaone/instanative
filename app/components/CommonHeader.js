import React, { Component } from 'react'
import { 
    View, 
    Image, 
    Text, 
    StyleSheet, 
    Dimensions, 
    TouchableOpacity 
} from 'react-native'

const styles = {
    commonHeader : {
        paddingTop: 20, 
        height: 62, 
        alignItems : "center", 
        justifyContent : "center", 
        backgroundColor : "#fff", 
        borderBottomWidth: 0.5, 
        borderBottomColor : "#ccc"
    }
}

/**
 * Bu komponent bütün ekranların en tepesindeki instagram logosunun olduğu header alanıdır.
 * Genel dizayn tasarlanırken ana amaç ortak olan kısımları komponent haline getirmektir 
 * böylece hem daha az kod yazılmış olur hem de müdahale etmek gerektiği zaman tek bir
 * noktada müdahale edilebilir.
 */
export default class CommonHeader extends Component {

    render() {
        return <View style={styles.commonHeader}>
            <Image source={require("../images/Instagram_logo.svg.png")} style={{width: 120, height : 36}} />
        </View>
    }
}