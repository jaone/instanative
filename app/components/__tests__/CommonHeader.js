import React from 'react';
import renderer from 'react-test-renderer';

import CommonHeader from '../CommonHeader';

it('renders a CommonHeader using Snapshots', () => {
  
  const _getComponent = (props) => {
    return (
      renderer.create(
        <CommonHeader/>
      )
    )
  };
  
  let component = _getComponent();
  expect(component).toMatchSnapshot();
});