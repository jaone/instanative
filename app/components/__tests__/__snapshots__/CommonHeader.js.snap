// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a CommonHeader using Snapshots 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderBottomColor": "#ccc",
      "borderBottomWidth": 0.5,
      "height": 62,
      "justifyContent": "center",
      "paddingTop": 20,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../app/images/Instagram_logo.svg.png",
      }
    }
    style={
      Object {
        "height": 36,
        "width": 120,
      }
    }
  />
</View>
`;
